pkg/tests/common/appstudio.go

```diff
@@ -19,26 +23,35 @@ var (
 var _ = framework.CommonSuiteDescribe("Red Hat App Studio common E2E", func() {
 	defer g.GinkgoRecover()
 	commonController, err := commonCtrl.NewSuiteController()
-	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	Expect(err).NotTo(HaveOccurred())
 
 	g.Context("Argo CD", func() {
 		for _, component := range AppStudioComponents {
 			g.It(component+" status", func() {
 				componentStatus, err := commonController.GetAppStudioComponentStatus(component, AppStudioComponentsNamespace)
-				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				gomega.Expect(componentStatus.Health.Status).To(gomega.Equal(health.HealthStatusHealthy))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(componentStatus.Health.Status).To(Equal(health.HealthStatusHealthy))
 
 			})
 		}
 	})
+	defer g.GinkgoRecover()
 
 	g.Context("ClusterTasks:", func() {
-		for _, pipelineName := range AppStudioPipelinesNames {
-			g.It("Check if "+pipelineName+" clustertask is pre-created", func() {
-				p, err := commonController.GetClusterTask(pipelineName, PipelinesNamespace)
-				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				gomega.Expect(p.Name).To(gomega.Equal(pipelineName))
+		g.It("Check if AppStudio ClusterTasks are precreated", func() {
+			err := wait.PollImmediate(30*time.Second, 10*time.Minute, func() (done bool, err error) {
+				for _, clusterTaskName := range AppStudioClusterTaskNames {
+					clusterTask, err := commonController.GetClusterTask(clusterTaskName, PipelinesNamespace)
+					if err != nil {
+						return false, nil
+					}
+					if clusterTaskName != clusterTask.Name {
+						return false, fmt.Errorf("Cluster task have an unexpected name")
+					}
+				}
+				return true, nil
 			})
-		}
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
it would be worth adding an additional message why the test failed - that the expected `ClusterTasks` were not found.

Reply:
Ussually gomega will print the error message received from `wait.Pollinmediate`

Reply:
That's true, but the message will contain only the information that the wait reached the timeout, but the developer doesn't know what it was waiting for. This is what I was referring to in your "CI stories for Integration testing" document - the failure message should contain enough information for troubleshooting.

For this particular case, it would be better to move the `for` cycle outside of the `PollImmediate` function and poll for each of the ClusterTasks separately. In this way, you could then construct the message for example in a way:
```go
("The ClusterTask %s hasn't been found. The present ClusterTasks are: %v", clusterTaskName, clusterTaskList)
```

Reply:
@flacatus didn't you miss my last comment here?